fvm/handler/contract.go

```diff
@@ -61,8 +71,14 @@ func (h *ContractHandler) SetContract(address runtime.Address, name string, code
 	add := flow.Address(address)
 	h.lock.Lock()
 	defer h.lock.Unlock()
+
+	encodedCode, err := encodeToGzip(code)
+	if err != nil {
+		return errors.NewContractCompressionError(flow.Address(address), name, err)
+	}
+
 	uk := programs.ContractUpdateKey{Address: add, Name: name}
-	u := programs.ContractUpdate{ContractUpdateKey: uk, Code: code}
+	u := programs.ContractUpdate{ContractUpdateKey: uk, Code: encodedCode}
```

Comment:
Not sure how expensive these calls are: 
```
	var b bytes.Buffer
	gz := gzip.NewWriter(&b)
	...
	gz.Close()
```

but if compression logic was moved into `func (h *ContractHandler) Commit() ([]programs.ContractUpdateKey, error) ` the number of those calls could potentially be slightly reduced, as you could create a single writer to compress all contracts that are pending an update.

another thought that crossed my mind, is could we start a go routine to compress this and that make sure it finished in the `Commit` call?